HW11-12_decorators/2_context_manager.py

```diff
@@ -0,0 +1,35 @@
+# Context manager - if give a "r" to open() the mistake will be shown
+file = open("text.txt", "w")
+print("===========")
+
+try:
+    file.write("My first context manager")
+except Exception as e:
+    print(f"An error occurred while writing to the file: {e}")
+finally:
+    file.close()
+
+
+with open("text.txt", "r") as some_file:
+    print(some_file.read())
+    print("===========")
+
+# OPTION 2 - didn't really get how it supposed to work
+# class MyManager:
+#     def __enter__(self):
+#         print("==========")
+#         return 1
+#
+#     def __exit__(self, exc_type, exc_val, exc_tb):
+# #        if exc_type == ValueError:
+# #            print(f"{exc_val}: error was found")
+#         print("Exit")
```

Comment:
if `return True` here it will ignore the exception

If an exception is supplied, and the method wishes to suppress the exception (i.e., prevent it from being propagated), it should return a true value. Otherwise, the exception will be processed normally upon exit from this method.